velox/exec/benchmarks/ExchangeBenchmark.cpp

```diff
@@ -109,126 +120,154 @@ class ExchangeBenchmark : public VectorTestBase {
-        })});
+      expected = makeRowVector({makeFlatVector<int64_t>(1, [&](auto /*row*/) {
+        return vectors.size() * vectors[0]->size() * width * taskWidth;
+      })});
+
+      // plan: Agg/kSingle(1) <-- Exchange (0)
+      plan =
+          exec::test::PlanBuilder()
+              .exchange(finalAggPlan->outputType(), VectorSerde::Kind::kPresto)
+              .singleAggregation({}, {"sum(a0)"})
+              .planNode();
+    };
 
     exec::test::AssertQueryBuilder(plan)
         .splits(finalAggSplits)
         .assertResults(expected);
-    auto elapsed = getCurrentTimeMicro() - startMicros;
-    int64_t bytes = 0;
-    int64_t repartitionNanos = 0;
-    int64_t exchangeNanos = 0;
-    int64_t exchangeBatches = 0;
-    int64_t exchangeRows = 0;
-    for (auto& task : tasks) {
-      auto stats = task->taskStats();
-      for (auto& pipeline : stats.pipelineStats) {
-        for (auto& op : pipeline.operatorStats) {
-          if (op.operatorType == "PartitionedOutput") {
-            repartitionNanos +=
-                op.addInputTiming.cpuNanos + op.getOutputTiming.cpuNanos;
-          } else if (op.operatorType == "Exchange") {
-            bytes += op.rawInputBytes;
-            exchangeRows += op.outputPositions;
-            exchangeBatches += op.outputVectors;
-            exchangeNanos +=
-                op.addInputTiming.cpuNanos + op.getOutputTiming.cpuNanos;
-          }
-        }
+
+    BENCHMARK_SUSPEND {
+      auto elapsed = getCurrentTimeMicro() - startMicros;
+      std::vector<int64_t> taskWallMs;
+
+      for (auto& task : leafTasks) {
+        auto taskStats = task->taskStats();
+        taskWallMs.push_back(
+            taskStats.executionEndTimeMs - taskStats.executionStartTimeMs);
+        auto planStats = toPlanStats(taskStats);
+        auto& taskPartitionedOutputStats =
+            planStats.at(leafPartitionedOutputId);
+        partitionedOutputStats += taskPartitionedOutputStats;
       }
-    }
 
-    counters.bytes += bytes;
-    counters.rows += width * vectors.size() * vectors[0]->size();
-    counters.usec += elapsed;
-    counters.repartitionNanos += repartitionNanos;
-    counters.exchangeNanos += exchangeNanos;
-    counters.exchangeRows += exchangeRows;
-    counters.exchangeBatches += exchangeBatches;
+      for (auto& task : finalAggTasks) {
+        auto taskStats = task->taskStats();
+        taskWallMs.push_back(
+            taskStats.executionEndTimeMs - taskStats.executionStartTimeMs);
+        auto planStats = toPlanStats(taskStats);
+
+        auto& taskPartitionedOutputStats =
+            planStats.at(finalAggPeartitionedOutputId);
+        partitionedOutputStats += taskPartitionedOutputStats;
+        auto repartitionRuntimeStats = taskPartitionedOutputStats.customStats;
+
+        auto& taskExchangeStats = planStats.at(exchangeId);
+        exchangeStats += taskExchangeStats;
+        auto exchangeRuntimeStats = taskExchangeStats.customStats;
```

Comment:
'exchangeRuntimeStats' seems not in use, would you confirm?

Reply:
Oh yes, I forgot to remove it. The content of it was mostly covered in the operator stats except the `compressedBytes` metrics. This benchmark doesn't include compressed tests so far, so I'll remove it for now. In the future we will need to add tests for compressed data.